src/ripple/core/sql_backend/SQLInterface_sqlite.cpp

```diff
@@ -0,0 +1,1280 @@
+            res = ripple::getMaxLedgerSeq(session, type);
+            return res ? false : true;
+        });
+    return res;
+}
+
+void
+SQLInterface_sqlite::deleteByLedgerSeq(
+    DatabaseType dtype,
+    TableType type,
+    LedgerIndex ledgerSeq)
+{
+    /* if databases exists, use it */
+    if (exists(dtype))
+    {
+        auto db = checkoutDb(dtype);
+        ripple::deleteByLedgerSeq(*db, type, ledgerSeq);
+        return;
+    }
+
+    /* else use shard database */
+    do_single(dtype, ledgerSeq, [&](soci::session& session, LedgerIndex index) {
+        ripple::deleteByLedgerSeq(session, type, ledgerSeq);
+        return true;
+    });
+}
+
+void
+SQLInterface_sqlite::deleteBeforeLedgerSeq(
+    DatabaseType dtype,
+    TableType type,
+    LedgerIndex ledgerSeq)
+{
+    /* if databases exists, use it */
+    if (exists(dtype))
+    {
+        auto db = checkoutDb(dtype);
+        ripple::deleteBeforeLedgerSeq(*db, type, ledgerSeq);
+        return;
+    }
+
+    /* else use shard databases */
+    iterate_back(
+        dtype,
+        seqToShardIndex(ledgerSeq),
+        [&](soci::session& session, LedgerIndex index) {
+            ripple::deleteBeforeLedgerSeq(session, type, ledgerSeq);
+            return true;
+        });
+}
+
+int
+SQLInterface_sqlite::getRows(DatabaseType dtype, TableType type)
+{
+    /* if databases exists, use it */
+    if (exists(dtype))
+    {
+        auto db = checkoutDb(dtype);
+        return ripple::getRows(*db, type);
+    }
+
+    /* else use shard databases */
+    int rows = 0;
+    iterate_forward(
+        dtype, (LedgerIndex)-1, [&](soci::session& session, LedgerIndex index) {
+            rows += ripple::getRows(session, type);
+            return true;
+        });
+    return rows;
+}
+
+std::tuple<int, int, int>
+SQLInterface_sqlite::getRowsMinMax(DatabaseType dtype, TableType type)
+{
+    /* if databases exists, use it */
+    if (exists(dtype))
+    {
+        auto db = checkoutDb(dtype);
+        return ripple::getRowsMinMax(*db, type);
+    }
+
+    /* else use shard databases */
+    int rows = 0, first = -1, last = -1;
+    iterate_forward(
+        dtype, (LedgerIndex)-1, [&](soci::session& session, LedgerIndex index) {
+            auto [r, f, l] = ripple::getRowsMinMax(session, type);
+            if (r)
+            {
+                rows += r;
+                if (first == -1)
+                    first = f;
+                last = l;
+            }
+            return true;
+        });
+    return {rows, first, last};
+}
+
+bool
+SQLInterface_sqlite::saveValidatedLedger(
+    Application& app,
+    std::shared_ptr<Ledger const> const& ledger,
+    bool current)
+{
+    /* if databases exists, use it */
+    if (exists(LEDGER) && exists(TRANSACTION))
+    {
+        return ripple::saveValidatedLedger(
+            *lgrdb_, *txdb_, app, ledger, current);
+    }
+
+    /* else use shard databases */
+#if 0
```

Comment:
Why is this commented out? Either remove it and put it in another pr or document why it's not enabled.

Reply:
Done.